### PR TITLE
Created User registration, working validation & acceptance tests

### PR DESCRIPTION
**Pull request by jgendrano**
Completed user signs up user story.

"Code checking" services are not merged to this branch so they are not available on this branch yet.

### CODE REVIEW

**Comment by jgendrano** [comment]
Ah, we assumed that since the badges weren't in the readme, those services wouldn't check this branch.

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+  # session. If you need permissions, you should implement that in a before filter.
+  # You can also supply a hash where the value is a boolean determining whether
+  # or not authentication should be aborted when the value is not present.
+  # config.authentication_keys = [:email]
+
+  # Configure parameters from the request object used for authentication. Each entry
+  # given should be a request method and it will automatically be passed to the
+  # find_for_authentication method and considered in your model lookup. For instance,
+  # if you set :request_keys to [:subdomain], :subdomain will be used on authentication.
+  # The same considerations mentioned for authentication_keys also apply to request_keys.
```
Comment (itiswicked) [inline config/initializers/devise.rb]: Delete comments!

File: db/migrate/20160105155415_devise_create_users.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      t.integer  :sign_in_count, default: 0, null: false
+      t.datetime :current_sign_in_at
+      t.datetime :last_sign_in_at
+      t.inet     :current_sign_in_ip
+      t.inet     :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
```
Comment (itiswicked) [inline db/migrate/20160105155415_devise_create_users.rb]: delete these comments

**Comment by itiswicked** [comment]
We're not sure what Hound wants us to do at this point.

**Pull request merged**